src/compute-client/src/protocol/command.rs

```diff
@@ -436,9 +436,7 @@ impl ComputeParameters {
 
     /// Return whether all parameters are unset.
     pub fn all_unset(&self) -> bool {
-        self.max_result_size.is_none()
-            && self.grpc_client.all_unset()
-            && self.dyncfg_updates.updates.is_empty()
+        *self == Self::default()
```

Comment:
This looks dangerous, what if the default changes? I think this method should check what it says on the tin even if it is more verbose.

Reply:
It's a good point, except that doing it the verbose way is also dangerous because sometimes folk add new fields without thinking to update this method. Note that it was incorrect before since it didn't check `tracing`.

Ideally I'd want to enforce that the `Default` impl always returns the all-empty object. Maybe we need a new `Empty` trait for that.

Reply:
This change now already sneaked in through https://github.com/MaterializeInc/materialize/pull/28508.

We have https://github.com/MaterializeInc/materialize/issues/17650 to track improving the various cluster parameter types, I think fixing the "all unset" semantics fits will into that scope.

Reply:
> is also dangerous because sometimes folk add new fields without thinking to update this method.

Yes that's a valid concern. An idiomatic way to guard against this is to add a match over the fields that doesn't have a wildcard match at the end (see how the `update` method does in the same struct). This way the code will fail to compile if the fields change.